docs/tutorials/on-device-training.md

```diff
@@ -0,0 +1,1406 @@
+   frozen_params = [
+      param.name
+      for param in onnx_model.graph.initializer
+      if param.name not in requires_grad
+   ]
+    ```
+
+3. Generate the training artifacts. We will use the `CrossEntropyLoss` loss and the `AdamW` optimizer for this tutorial.
+
+    ```python
+   from onnxruntime.training import artifacts
+
+   # Generate the training artifacts.
+   artifacts.generate_artifacts(
+      onnx_model,
+      requires_grad=requires_grad,
+      frozen_params=frozen_params,
+      loss=artifacts.LossType.CrossEntropyLoss,
+      optimizer=artifacts.OptimType.AdamW,
+      artifact_directory="training_artifacts"
+   )
+    ```
+
+4. That's all. The training artifacts have been generated in the `training_artifacts` folder. This marks the end of the offline phase. These artifacts are ready to be deployed to the Android device for training.
+
+## Training Phase - Android application development
+
+1. Set up the project in Android Studio
+
+   a. Open Android Studio and click `New Project`
+   ![Android Studio Setup - New Project](../../images/on-device-training-android-studio-setup.png)
+
+   b. Click on `Native C++` -> `Next`. Fill out the `New Project` details as follows:
+
+     - Name - `ORT Personalize`
+     - Package Name - `com.example.ortpersonalize`
+     - Language - `Kotlin`
+
+      Click `Next`.
+
+   ![Android Studio Setup - Project Name](../../images/on-device-training-android-studio-project-name.png)
+
+   c. Select the `C++17` toolchain -> `Finish`
+
+   ![Android Studio Setup - Project C++ ToolChain](../../images/on-device-training-android-studio-toolchain.png)
+
+   d. That's it. The Android Studio project has been set up. You should now be able to see the Android Studio editor with some boiler plate code.
+
+2. Add the ONNX Runtime dependency
+
+   a. Create two new folder called `lib` and `include\onnxruntime` under the cpp directory in the Android Studio project.
+
+   ![lib and include folder](../../images/on-device-training-lib-include.png)
+
+   b. Head over to [Maven Central](https://central.sonatype.com/artifact/com.microsoft.onnxruntime/onnxruntime-training-android/) and download the `onnxruntime-training-android` archive package (aar file).
+
+   c. Rename the `aar` extension to `zip`. So `onnxruntime-training-android-1.15.0.aar` becomes `onnxruntime-training-android-1.15.0.zip`.
+
+   d. Extract the contents of the zip file.
+
+   e. Copy the `libonnxruntime.so` shared library from the `jni\arm64-v8a` folder to your Android project under the newly created `lib` folder.
+
+   f. Copy the contents of the `headers` folder to the newly created `include\onnxruntime` folder.
+
+   g. In the `native-lib.cpp` file, include the training cxx header file.
+
+   ```cpp
+   #include "onnxruntime_training_cxx_api.h"
+   ```
+
+   h. Add `abiFilters` to the `build.gradle (Module)` file so as to select `arm64-v8a`. This setting must be added under `defaultConfig` in `build.gradle`:
```

Comment:
For this step, maybe including the core for the whole build.gradle module and adding the diff with the ndk lines would look better than just including the new lines. This way people know the code before/after the code you want then to insert.